src/applications/vaos/project-cheetah/components/SelectDate2Page.jsx

```diff
@@ -1,46 +1,238 @@
+  selectedDate,
+  timezoneDescription,
+  availableSlots = [],
+) {
+  return availableSlots.reduce((acc, slot) => {
+    if (slot.start.split('T')[0] === selectedDate) {
+      let time = moment(slot.start);
+      if (slot.start.endsWith('Z')) {
+        time = time.tz(timezoneDescription);
+      }
+      const meridiem = time.format('A');
+      const screenReaderMeridiem = meridiem.replace(/\./g, '').toUpperCase();
+      acc.push({
+        value: slot.start,
+        label: (
+          <>
+            {time.format('h:mm')} <span aria-hidden="true">{meridiem}</span>{' '}
+            <span className="sr-only">{screenReaderMeridiem}</span>
+          </>
+        ),
+      });
+    }
+    return acc;
+  }, []);
+}
+
+function ErrorMessage({ facilityId, requestAppointmentDateChoice }) {
+  return (
+    <div aria-atomic="true" aria-live="assertive">
+      <AlertBox
+        status="error"
+        headline="We’ve run into a problem when trying to find available appointment times"
+      >
+        To schedule this appointment, you can{' '}
+        <button
+          onClick={() => requestAppointmentDateChoice(history)}
+          className="va-button-link"
+        >
+          submit a request for a VA appointment
+        </button>{' '}
+        or{' '}
+        <a
+          href={`/find-locations/facility/vha_${getRealFacilityId(facilityId)}`}
+          rel="noopener noreferrer"
+          target="_blank"
+        >
+          call your local VA medical center
+        </a>
+        .
+      </AlertBox>
+    </div>
+  );
+}
+
+function userSelectedSlot(calendarData) {
+  return calendarData?.selectedDates?.length > 0;
+}
+
+function goBack({ routeToPreviousAppointmentPage, history }) {
+  return routeToPreviousAppointmentPage(history, pageKey);
+}
+
+function validate({ calendarData, setValidationError }) {
+  if (userSelectedSlot(calendarData)) {
+    setValidationError(null);
+  } else {
+    setValidationError(missingDateError);
+  }
+}
+
+function goForward({
+  data,
+  history,
   routeToNextAppointmentPage,
+  submitted,
+  setSubmitted,
+  setValidationError,
+}) {
+  const { calendarData } = data || {};
+  validate({ calendarData, setValidationError });
+  if (userSelectedSlot(calendarData)) {
+    routeToNextAppointmentPage(history, pageKey);
+  } else if (submitted) {
+    scrollAndFocus('.usa-input-error-message');
+  } else {
+    setSubmitted(true);
+  }
+}
+
+export function SelectDate2Page({
+  appointmentSlotsStatus,
+  availableDates,
+  availableSlots,
+  data,
+  facilityId,
+  getAppointmentSlots,
+  pageChangeInProgress,
+  onCalendar2Change,
   routeToPreviousAppointmentPage,
+  requestAppointmentDateChoice,
+  routeToNextAppointmentPage,
+  timezone,
+  timezoneDescription,
 }) {
   const history = useHistory();
+  const [submitted, setSubmitted] = useState(false);
+  const [validationError, setValidationError] = useState(null);
+
   useEffect(() => {
+    getAppointmentSlots(
+      moment()
```

Comment:
Rather than using the current date to generate these dates, I think I would use the date chosen in the first calendar. I believe the logic here will fail if you try to schedule your first appointment a couple months in advance.

The `startOf`, `add`, and `endOf` adjustments are all good to keep, I think just starting from the previously chosen date will make things more reliable.

Reply:
Good call, now that I fixed the naming collision it's trivial. 👍